### DISCUSSION TITLE
Example about reading and sending mails in office 365 mailbox

### DISCUSSION THREAD

**Discussion by 44640766** [microsoftgraph/msgraph-sdk-go#154] [Q&A] [answered]
Hi,
     I want to use msgraph-sdk-go to read and send mails. My mailbox is on office 365. However there are lots of APIs in msgraph-sdk-go, I don't know which one can work. Does anybody provide an example for reading or sending mails in office 365 mailbox?
     Thanks very much.

**Comment by baywet** [comment, marked as answer]
Did you see the documentation [for sendmail](https://docs.microsoft.com/en-us/graph/api/user-sendmail?view=graph-rest-1.0&tabs=go) and for [read mail](https://docs.microsoft.com/en-us/graph/api/user-list-messages?view=graph-rest-1.0&tabs=go) ?

**Reply by 44640766** [reply]
Hi, Baywet,
     Nice to get answer from you:) I saw the two documentation you linked. But I can't run them successful unfortunately. I am a new guy both for Golang and Msgraph.  It will be great help if you can correct me. Thanks very much.
     My demo codes:

	cred, err := azidentity.NewClientSecretCredential(
		"****",
		"*****",
		"*****",
		nil,
	)
	if err != nil {
		fmt.Println(err)
	}
	auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(cred, []string{"IMAP.AccessAsUser.All/.default"})
	if err != nil {
		fmt.Println("1:", err)
	}
	adapter, err := msgraphsdk.NewGraphRequestAdapter(auth)
	if err != nil {
		fmt.Println("2:", err)
	}
	client := msgraphsdk.NewGraphServiceClient(adapter)

	result, err := client.Me().Messages().Get()
	//result, err := client.Me().Get()
	if err != nil {
		fmt.Println("3:", err)
	}

error_description": "AADSTS70011: The provided request must include a 'scope' input parameter. The provided value for the input parameter 'scope' is not valid. The scope IMAP.AccessAsUser.All/.default openid offline_access profile is not valid.

![image](https://user-images.githubusercontent.com/37007411/166923350-2fceea05-a862-402f-bb4a-d2a5f42b0252.png)

**Reply by baywet** [reply]
The IMAP and SMTP scopes are reserved to applications communicating with the mailbox over IMAP/SMTP. These scopes don't apply to your application since you're using Microsoft Graph and communicating via HTTP/REST.
```diff
-auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(cred, []string{"IMAP.AccessAsUser.All/.default"})
+auth, err := a.NewAzureIdentityAuthenticationProviderWithScopes(cred, []string{"Mail.ReadWrite", "Mail.Send"})
```

**Reply by 44640766** [reply]
thank you very much.
